src/Symfony/Component/HttpClient/Response/AmpResponse.php

```diff
@@ -278,7 +279,8 @@ private static function followRedirects(Request $originRequest, AmpClientState $
         }
 
         $originRequest->setBody(new AmpBody($options['body'], $info, $onProgress));
-        $response = $multi->request($options, $originRequest, $canceller->getCancellation(), $info, $onProgress, $handle);
+
+        $response = $multi->request($options, $originRequest, $canceller->getCancellation(), $info, $onProgress, $handle, $canceller);
```

Comment:
maybe we should change the signature to allow that (or use the cancellation directly if that makes sense)?
```suggestion
        $response = $multi->request($options, $originRequest, $canceller, $info, $onProgress, $handle);
```